// @myparcel/js-sdk/dist/schema/private/address.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.update = exports.create = void 0;
var create = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'address_create',
  type: 'object',
  required: ['data'],
  additionalProperties: false,
  properties: {
    data: {
      type: 'object',
      additionalProperties: false,
      required: ['addresses'],
      properties: {
        addresses: {
          type: 'array',
          minItems: 1,
          items: {
            type: 'object',
            additionalProperties: false,
            required: ['account_id', 'cc', 'city', 'street', 'person'],
            properties: {
              account_id: {
                type: 'integer'
              },
              shop_id: {
                type: 'integer'
              },
              cc: {
                $ref: 'shared#/definitions/country-code'
              },
              region: {
                type: 'string',
                maxLength: 35
              },
              postal_code: {
                type: 'string'
              },
              city: {
                type: 'string'
              },
              street: {
                type: 'string'
              },
              street_additional_info: {
                type: 'string'
              },
              number: {
                type: 'string'
              },
              number_suffix: {
                type: 'string'
              },
              area: {
                type: 'string',
                maxLength: 35
              },
              person: {
                type: 'string'
              },
              company: {
                type: 'string'
              },
              email: {
                type: 'string',
                anyOf: [{
                  type: 'string',
                  format: 'email'
                }, {
                  type: 'string',
                  maxLength: 0
                }]
              },
              phone: {
                type: 'string'
              }
            }
          }
        }
      }
    }
  }
};
exports.create = create;
var update = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'address_update',
  type: 'object',
  required: ['data'],
  additionalProperties: false,
  properties: {
    data: {
      type: 'object',
      additionalProperties: false,
      required: ['addresses'],
      properties: {
        addresses: {
          type: 'array',
          minItems: 1,
          items: {
            type: 'object',
            additionalProperties: false,
            required: ['id', 'account_id', 'cc', 'city', 'street', 'person'],
            properties: {
              id: {
                type: 'integer'
              },
              account_id: {
                type: 'integer'
              },
              shop_id: {
                type: 'integer'
              },
              cc: {
                $ref: 'shared#/definitions/country-code'
              },
              region: {
                type: 'string',
                maxLength: 35
              },
              postal_code: {
                type: 'string'
              },
              city: {
                type: 'string'
              },
              street: {
                type: 'string'
              },
              street_additional_info: {
                type: 'string'
              },
              number: {
                type: 'string'
              },
              number_suffix: {
                type: 'string'
              },
              area: {
                type: 'string',
                maxLength: 35
              },
              person: {
                type: 'string'
              },
              company: {
                type: 'string'
              },
              email: {
                type: 'string',
                anyOf: [{
                  type: 'string',
                  format: 'email'
                }, {
                  type: 'string',
                  maxLength: 0
                }]
              },
              phone: {
                type: 'string'
              }
            }
          }
        }
      }
    }
  }
};
exports.update = update;
var _default = {
  create: create,
  update: update
};
exports.default = _default;
