// @myparcel/js-sdk/dist/schema/private/collect-trip-schedule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var create = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'collect-trip-schedule_create',
  type: 'object',
  required: ['data'],
  properties: {
    data: {
      type: 'object',
      properties: {
        collect_trip_schedule: {
          type: 'array',
          items: {
            additionalProperties: false,
            type: 'object',
            required: ['carrier_id', 'collect_window_start_time', 'contact', 'day_of_the_week', 'planning_start_date', 'shop_id'],
            properties: {
              carrier_id: {
                type: 'integer'
              },
              collect_window_start_time: {
                type: 'string',
                maxLength: 255
              },
              collect_window_end_time: {
                type: 'string',
                maxLength: 255
              },
              contact: {
                required: ['phone', 'street', 'number', 'postal_code', 'cc'],
                properties: {
                  cc: {
                    type: 'string',
                    minLength: 2,
                    maxLength: 2
                  },
                  city: {
                    type: 'string',
                    maxLength: 255
                  },
                  company: {
                    type: 'string',
                    maxLength: 255
                  },
                  email: {
                    type: 'string',
                    format: 'email'
                  },
                  first_name: {
                    type: 'string',
                    maxLength: 255
                  },
                  gender: {
                    type: 'string',
                    enum: ['mr', 'mrs']
                  },
                  last_name: {
                    type: 'string',
                    maxLength: 255
                  },
                  number: {
                    type: ['integer', 'string'],
                    maxLength: 255
                  },
                  number_suffix: {
                    type: 'string',
                    maxLength: 255
                  },
                  phone: {
                    type: 'string',
                    minLength: 8,
                    maxLength: 255
                  },
                  postal_code: {
                    type: 'string',
                    maxLength: 255
                  },
                  street: {
                    type: 'string',
                    maxLength: 255
                  },
                  street_additional_info: {
                    type: 'string',
                    maxLength: 255
                  }
                },
                type: 'object'
              },
              day_of_the_week: {
                type: 'integer',
                minimum: 1,
                maximum: 5
              },
              package_count: {
                type: 'integer',
                multipleOf: 5,
                minimum: 5,
                maximum: 25
              },
              planning_start_date: {
                $ref: 'shared#/definitions/date'
              },
              planning_end_date: {
                $ref: 'shared#/definitions/date'
              },
              shop_id: {
                type: 'integer',
                maxLength: 255
              },
              volume: {
                type: 'integer',
                minimum: 1
              },
              volume_type: {
                type: 'string',
                enum: ['package', 'mailbag', 'container', 'pallet']
              },
              courier_product_id: {
                type: 'integer'
              }
            }
          }
        }
      }
    }
  }
};
var update = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'collect-trip-schedule_update',
  properties: {
    data: {
      properties: {
        collect_trip_schedule: {
          items: {
            additionalProperties: true,
            properties: {
              id: {
                type: 'integer'
              },
              carrier_id: {
                type: 'integer'
              },
              collect_window_start_time: {
                type: 'string',
                maxLength: 255
              },
              collect_window_end_time: {
                type: 'string',
                maxLength: 255
              },
              contact: {
                properties: {
                  cc: {
                    type: 'string',
                    minLength: 2,
                    maxLength: 2
                  },
                  city: {
                    type: 'string',
                    maxLength: 255
                  },
                  company: {
                    type: 'string',
                    maxLength: 255
                  },
                  email: {
                    type: 'string',
                    format: 'email'
                  },
                  first_name: {
                    type: 'string',
                    maxLength: 255
                  },
                  gender: {
                    type: 'string',
                    enum: ['mr', 'mrs']
                  },
                  last_name: {
                    type: 'string',
                    maxLength: 255
                  },
                  number: {
                    type: ['integer', 'string'],
                    maxLength: 255
                  },
                  number_suffix: {
                    type: 'string',
                    maxLength: 255
                  },
                  phone: {
                    type: 'string',
                    minLength: 8,
                    maxLength: 255
                  },
                  postal_code: {
                    type: 'string',
                    maxLength: 255
                  },
                  street: {
                    type: 'string',
                    maxLength: 255
                  },
                  street_additional_info: {
                    type: 'string',
                    maxLength: 255
                  }
                },
                type: 'object'
              },
              day_of_the_week: {
                type: 'integer',
                minimum: 1,
                maximum: 5
              },
              package_count: {
                type: 'integer',
                multipleOf: 5,
                minimum: 5,
                maximum: 25
              },
              planning_start_date: {
                $ref: 'shared#/definitions/date'
              },
              planning_end_date: {
                $ref: 'shared#/definitions/date'
              },
              shop_id: {
                type: 'integer'
              },
              volume: {
                type: 'integer',
                minimum: 1
              },
              volume_type: {
                type: 'string',
                enum: ['package', 'mailbag', 'container', 'pallet']
              },
              courier_product_id: {
                type: 'integer'
              }
            },
            type: 'object'
          },
          type: 'array'
        }
      },
      type: 'object'
    }
  },
  type: 'object'
};
var _default = {
  create: create,
  update: update
};
exports.default = _default;
