// @myparcel/js-sdk/dist/schema/private/return-contribution.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var create = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'return-contribution_create',
  type: 'object',
  required: ['data'],
  additionalProperties: false,
  properties: {
    data: {
      type: 'object',
      additionalProperties: false,
      required: ['return_contributions'],
      properties: {
        return_contributions: {
          type: 'array',
          minItems: 1,
          items: {
            type: 'object',
            additionalProperties: false,
            required: ['shop_id', 'cc'],
            properties: {
              shop_id: {
                type: 'number'
              },
              package_type: {
                type: 'number',
                enum: [1, 2]
              },
              cc: {
                $ref: 'shared#/definitions/country-code'
              },
              price: {
                allOf: [{
                  $ref: 'shared#/definitions/price'
                }, {
                  properties: {
                    amount: {
                      min: 0,
                      max: 10000
                    }
                  }
                }]
              },
              position_order: {
                type: 'number'
              }
            }
          }
        }
      }
    }
  }
};
var update = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'return-contribution_update',
  type: 'object',
  required: ['data'],
  additionalProperties: false,
  properties: {
    data: {
      type: 'object',
      additionalProperties: false,
      required: ['return_contributions'],
      properties: {
        return_contributions: {
          type: 'array',
          minItems: 1,
          items: {
            type: 'object',
            additionalProperties: false,
            required: ['id', 'shop_id', 'cc'],
            properties: {
              id: {
                type: 'number'
              },
              shop_id: {
                type: 'number'
              },
              package_type: {
                type: 'number',
                enum: [1, 2]
              },
              cc: {
                $ref: 'shared#/definitions/country-code'
              },
              price: {
                allOf: [{
                  $ref: 'shared#/definitions/price'
                }, {
                  properties: {
                    amount: {
                      min: 0,
                      max: 10000
                    }
                  }
                }]
              },
              position_order: {
                type: 'number'
              }
            }
          }
        }
      }
    }
  }
};
var _default = {
  create: create,
  update: update
};
exports.default = _default;
