// @myparcel/js-sdk/dist/action/get.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _nodeFetch = _interopRequireWildcard(require("node-fetch"));

var _transform = _interopRequireDefault(require("@teamawesome/transform"));

var _default = function _default(superclass) {
  var _class, _temp;

  return _temp = _class =
  /*#__PURE__*/
  function (_superclass) {
    (0, _inherits2.default)(Get, _superclass);

    function Get() {
      (0, _classCallCheck2.default)(this, Get);
      return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Get).apply(this, arguments));
    }

    (0, _createClass2.default)(Get, [{
      key: "get",

      /**
       * @type {Pipe}
       */

      /**
       * @param {number} id
       */
      value: function get(id) {
        return this.constructor.get.call(id, this);
      }
    }]);
    return Get;
  }(superclass), (0, _defineProperty2.default)(_class, "get", new _transform.default(get)), _temp;
};

exports.default = _default;
var get = new _transform.default({
  url: function url(id) {
    var _this$config$url = this.config.url,
        origin = _this$config$url.origin,
        pathname = _this$config$url.pathname;
    return new URL("".concat(origin).concat(pathname, "/").concat(this.endpoint, "/").concat(id));
  },
  request: function request(url) {
    var request = {
      headers: {
        accept: 'application/json;charset=utf-8',
        'content-type': 'application/json;charset=utf-8'
      }
    };

    if (this.config.token) {
      request.headers.authorization = "basic ".concat(this.config.token);
    }

    if (this.config.acceptLanguage) {
      request.headers['Accept-Language'] = this.config.acceptLanguage;
    }

    return new _nodeFetch.Request(url, request);
  },
  response: function response(request) {
    return (0, _nodeFetch.default)(request);
  },
  parse: function parse(response) {
    var _this = this;

    return response.json().then(function (data) {
      if (response.ok) {
        return data.data[_this.namespace][0];
      }

      return Promise.reject(data);
    });
  }
});
